internal/runner/runner.go

```diff
@@ -60,68 +58,64 @@ func (r *runner) Run(ctx context.Context) error {
 		return err
 	}
 
-	namespaceList, err := r.cs.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	// get list of pods from all namespaces
+	podList, err := r.cs.CoreV1().Pods("").List(cancelCtx, metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	for _, ns := range namespaceList.Items {
-		podList, err := r.cs.CoreV1().Pods(ns.Name).List(ctx, metav1.ListOptions{})
-		if err != nil {
-			return err
-		}
-
-		for _, pod := range podList.Items {
+	for _, pod := range podList.Items {
+		go r.streamPodLogs(cancelCtx, r.cs, pod)
 
-			for _, container := range pod.Spec.Containers {
-				req := r.cs.CoreV1().Pods(ns.Name).GetLogs(pod.Name, &corev1.PodLogOptions{
-					Timestamps: true,
-					Container:  container.Name,
-				})
+	}
 
-				podLogs, err := req.Stream(ctx)
-				if err != nil {
-					return err
-				}
-				defer podLogs.Close()
+	return nil
+}
 
-				buf := new(bytes.Buffer)
+func (r *runner) loadLogsToLoki(logLine string, logParser parser.Parser, labels map[string]string) error {
+	tm, labelset, err := logParser.Parse(logLine, labels)
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		r.logger.Log("Skipping log due to invalid parse", "Error", err.Error())
+		return err
+	}
+	r.lokiClient.PostLog(logLine, tm, labelset)
 
-				_, err = io.Copy(buf, podLogs)
-				if err != nil {
-					return err
-				}
+	return nil
+}
 
-				labels := make(map[string]string)
-				labels["namespace"] = ns.Name
-				labels["pod_name"] = pod.Name
-				labels["container_name"] = container.Name
+// streamPodLogs will stream the pod logs and load the logs to loki with relevant
+// labels, loglines and timestamp
+func (r *runner) streamPodLogs(cancelCtx context.Context, cs kubernetes.Interface, pod corev1.Pod) error {
```

Comment:
@zawachte I want to know if I can utilize the `cancelCtx` here to return `cancelCtx.Done()` when the context deadline is exceeded when `streamPodLogs` is called. I am still catching up with the goroutine channels aspect. Is there any suggestions you have for me to handle this better?